smoke/src/main/java/com/adobe/cq/cloud/testing/it/smoke/HtmlUnitClient.java

```diff
@@ -165,4 +166,30 @@ private void login(WebClient webClient) throws ClientException {
         }
     }
 
+    /**
+     * Extracts URI reference from specified element and converts it to URI
+     * This method will trigger junit assertion if refAttr value cannot be parsed as URI
+     * providing comprehensive error message.
+     * @param node - html element from which to extract reference
+     * @param refAttr - name of the attribute containing corresponding value
+     * @return refAttr value as URI or <code>null</code> if attribute does not exist
+     */
+    private static URI getNamedItemAsUri(DomNode node, String refAttr) {
+        Node src = node.getAttributes().getNamedItem(refAttr);
+        if (src == null) {
+            return null;
+        } else {
+            try {
+                String href = src.getNodeValue();
+                return new URI(href);
+            } catch (URISyntaxException e) {
+                fail("Invalid URI value in [" + refAttr + "] attribute in: [" + node + "].\n" +
+                        "   Page URL:  [" + node.getPage().getUrl() + "]\n" +
+                        "   XPath:     [" + node.getCanonicalXPath() + "]\n" +
+                        "   Caused by: [" + e.getMessage() + "]");
+                throw new AssertionError(); // must never happen
```

Comment:
then why is it here? 😛

Reply:
It is to make the compiler happy.
The compiler requires each code branch in non-void method to return a value or throw an exception.
So the choice is between returning null or throwing something. I chose throwing because it acts as a fail fast validation of above reasoning.